### ISSUE TITLE
sam local tests don't work

### ISSUE THREAD

**Issue by otterley** [aws-cloudformation/cloudformation-cli-go-plugin#116]
I'm following the [documentation](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-walkthrough.html#resource-type-walkthrough-test) for running tests after building my code. 

I've created an event JSON file that looks like this:

```json
{
    "credentials": {
        "accessKeyId": "xxx",
        "secretAccessKey": "xxx",
        "sessionToken": "xxx"
    },
    "action": "CREATE",
    "request": {
        "clientRequestToken": "4b90a7e4-b790-456b-a937-0cfdfa211dfe",
        "desiredResourceState": {
            "ClusterName": "test",
            "Namespace": "test-namespace",
            "ServiceAccountName": "test-svcacct",
            "Description": "Test Role, assumed by test-svcacct in test-namespace",
            "ManagedPolicyArns": [
                "arn:aws:iam::aws:policy/ReadOnlyAccess"
            ]
        },
        "logicalResourceIdentifier": "MyRole"
    },
    "callbackContext": null
}
```

(The credentials are valid.)

When I try to invoke the function via `sam local invoke` as directed in the documentation, I receive the following error:

```
$ sam local invoke TestEntrypoint --event sam-tests/delete.json
Invoking handler (go1.x)

Fetching lambci/lambda:go1.x Docker container image......
Mounting /xxx/bin as /var/task:ro,delegated inside runtime container
START RequestId: a629ee8a-426e-11f9-ff85-38046f97bf3c Version: $LATEST
END RequestId: a629ee8a-426e-11f9-ff85-38046f97bf3c
REPORT RequestId: a629ee8a-426e-11f9-ff85-38046f97bf3c  Init Duration: 303.87 ms        Duration: 18.19 ms      Billed Duration: 100 ms Memory Size: 128 MB     Max Memory Used: 25 MB

{"errorType":"baseError","errorMessage":"Validation: Failed Validation\ncaused by: ResponseEndpoint: zero value, ResourceType: zero value, BearerToken: zero value, Region: zero value"}
```

Any idea what might be going on here?

**Comment by stlava** [comment]
Running into the same issue here.

**Comment by brianterry** [comment]
The latest version of this plugin supports contract test and sam local test. The update has been pushed to the master branch; however, it needs a release tag in order for go modules to pull it down. 

For now, follow these steps to enable sam local and contract test: 
1. Update your project to use the master branch
`go get github.com/aws-cloudformation/cloudformation-cli-go-plugin@master`

2. Update the `template.yml` file by adding an environment variable to the TestEntrypoint
```
    TestEntrypoint:
    Type: AWS::Serverless::Function
    Properties:
      Handler: handler
      Runtime: go1.x
      CodeUri: bin/
      Environment: 
       Variables: 
         MODE: Test
```

**Comment by brianterry** [comment]
Update: This issue was fixed in the 1.13 release:
https://github.com/aws-cloudformation/cloudformation-cli-go-plugin/releases/tag/v0.1.3

Closing this issue.
